src/psyclone/tests/psyGen_test.py

```diff
@@ -2617,8 +2627,269 @@ def test_binaryoperation_can_be_printed():
+        real(wp), dimension(:,:), intent(in)  :: f1
+        real(wp), dimension(:,:), intent(out)  :: f2
+        real(wp), dimension(:,:), intent(inout)  :: f3
+        f2 = f1 + 1
+      end subroutine dummy_code
+    end module dummy_mod
+    '''
+    ast1 = fpapi.parse(DUMMY_KERNEL_METADATA, ignore_comments=True)
+    metadata = DynKernMetadata(ast1)
+    my_kern = DynKern()
+    my_kern.load_meta(metadata)
+    ast2 = my_kern.ast
+
+    # Test properly formed kernel module
+    schedule = processor.generate_schedule("dummy_code", ast2)
+    assert isinstance(schedule, KernelSchedule)
+
+    # Test corrupting ast by deleting execution part
+    del ast2.content[0].content[2].content[1].content[2]
+    schedule = processor.generate_schedule("dummy_code", ast2)
+    assert isinstance(schedule, KernelSchedule)
+
+
+def test_fparser2astprocessor_process_declarations():
+    from fparser.two.parser import ParserFactory
+    from fparser.common.readfortran import FortranStringReader
+    from fparser.two.Fortran2003 import Specification_Part
+    ParserFactory().create(std="f2008")
+    fake_parent = KernelSchedule("dummy_schedule")
+    processor = Fparser2ASTProcessor()
+
+    # Test simple declarations
+    reader = FortranStringReader("integer :: l1")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("l1").name == 'l1'
+    assert fake_parent.symbol_table.lookup("l1").datatype == 'integer'
+    assert fake_parent.symbol_table.lookup("l1").dimensions == 0
+    assert fake_parent.symbol_table.lookup("l1").kind == 'local'
+
+    reader = FortranStringReader("Real      ::      l2")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("l2").name == "l2"
+    assert fake_parent.symbol_table.lookup("l2").datatype == 'real'
+    assert fake_parent.symbol_table.lookup("l2").dimensions == 0
+    assert fake_parent.symbol_table.lookup("l2").kind == 'local'
+
+    # Initialisations not supported yet
+    reader = FortranStringReader("integer :: l1 = 1")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    with pytest.raises(NotImplementedError) as error:
+        processor.process_declarations(fake_parent, [fparser2specification])
+
+    # Test different intent specifications
+    reader = FortranStringReader("integer, intent(in) :: arg1")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("arg1").name == "arg1"
+    assert fake_parent.symbol_table.lookup("arg1").datatype == 'integer'
+    assert fake_parent.symbol_table.lookup("arg1").dimensions == 0
+    assert fake_parent.symbol_table.lookup("arg1").kind == 'read_arg'
+
```

Comment:
Please break-out the following tests (for differing values of intent) into a separate test.

Reply:
Done